packages/components/src/disabled/README.md

```diff
@@ -1,6 +1,8 @@
 # Disabled
 
-Disabled is a component which disables descendant tabbable elements and prevents pointer interaction.
+Disabled is a component which disables descendant tabbable form input elements and prevents pointer interaction.
+
+Note: by default this component does not prevent tabbing to anchor elements. If this is required then you may utilise the `onDisable` prop to manually disable individual elements on a per node basis.
```

Comment:
It feels like a bug to me and that we should also disable these by default unless there's something I'm missing.

Reply:
It's a known issue.

For example, where in [this code](https://github.com/WordPress/gutenberg/pull/17064/files#diff-beb3ac40e8f8c08b7c848a5d7c78a803R74) does it disable anchors?

Reply:
My understanding of the component was that it should disable every tabbable element. Not sure who can confirm/inform this? cc @jasmussen @gziolo

Reply:
Personally I feel a `Disabled` component should disable all interactions. Whether or not this is an a11y concern is not an implementation detail of this component. Rather it is a concern **at the point at which the component is _used_**.

Therefore I propose we make the `Disabled` component completely disabled all interactions within the wrapped component.

We could add a delegated event listener on `this.node` and `preventDefault()` on all event types. Combined with the existing `pointer-events` approach here this would completely disabled interaction. 

Noting that we _might_ also need to add `tabindex="-1"` to add focusable nodes to ensure that they cannot receive focus in the first place, else calling `preventDefault` might lead to strange focus context loss.

What do you think @youknowriad?

Reply:
I agree with the reasoning especially if we take a look at the places this component is used. I feel we should disable everything. It's like applying a nested `role="presentation"`

Reply:
Ok @youknowriad I've updated the component to disable the ability to focus or tab to elements within the child component.